apps/src/StudioApp.js

```diff
@@ -722,38 +738,43 @@ StudioApp.prototype.onResize = function() {
   var headersHeight = parseInt(window.getComputedStyle(headers).height, 10);
 
   div.style.top = divParent.offsetTop + 'px';
-  var fullWorkspaceWidth = parentWidth - (gameWidth + WORKSPACE_PLAYSPACE_GAP);
-  var oldWidth = parseInt(div.style.width, 10) || div.getBoundingClientRect().width;
-  div.style.width = fullWorkspaceWidth + 'px';
 
-  // Keep blocks static relative to the right edge in RTL mode
-  if (this.isUsingBlockly() && Blockly.RTL && (fullWorkspaceWidth - oldWidth !== 0)) {
-    Blockly.mainBlockSpace.getTopBlocks().forEach(function(topBlock) {
-      topBlock.moveBy(fullWorkspaceWidth - oldWidth, 0);
-    });
-  }
+  var fullWorkspaceWidth = parentWidth;
+  if (this.usesStandardLayout_) {
```

Comment:
It seems like most of this method is handling the standard case. Would it make more sense to have your case override onResize to do the subset of things it wants?